mitodl/ocw-studio
# Rename mass-publish pipeline to mass-build-sites, refactor mass-publish command

#### Pre-Flight checklist

#### What are the relevant tickets?
Related to #1222
Closes #1203

#### What's this PR do?
- Renames the `mass-publish` pipeline to `mass-build-sites` and renames associated functions.
- Makes the query exclusion filter for the `mass_publish` command the same as for the `WebsitePublishViewSet` endpoint API, regardless of `source`.
- Adds an option to the management command to run individual site pipelines instead of the mass-build-sites pipeline.

#### How should this be manually tested?
- Follow README for setting up a local concourse instance
- Run `docker-compose --profile concourse up`
- Run `docker-compose run web python manage.py upsert_mass_build_pipeline`, it should create draft and live `mass-build-sites` pipelines
- If you don't already have individual site pipelines, make those too with `manage.py backpopulate_pipelines`.
- Run `docker-compose run web python manage.py mass_publish --no-mass-build draft` (optionally with `--filter`).  It should ask you if you really want to run individual concourse pipelines for X sites.  The number of sites should be consistent with the number of sites where `draft_publish_date` is not None (and your filter if any).
- Clicking `y` should cause the command to proceed, `n` to abort.
- Verify that the individual pipelines are started (they will never succeed so you can abort them).
- Run `docker-compose run web python manage.py mass_publish draft` (optionally with `--filter`).  You should not get a confirmation dialog.
- Verify that the mass-build-sites draft pipeline is started (it will fail so you can abort it).


The original `mass-publish` pipelines on RC and prod concourse should be deleted via `fly` soon before or after the new `mass-build-sites` pipelines are created
